### DISCUSSION TITLE
What is the logic behind --bbox?

### DISCUSSION THREAD

**Discussion by kuwapa** [systemed/tilemaker#343] [General]
I was trying to render just the capital region for australia and used this bounding box value - `148.769452,-35.919616,149.4044,-35.139509` (minY, minX, maxY, maxX). 

Used this command
`tilemaker --input australia-latest.osm.pbf --output capital.mbtiles --bbox 148.769452,-35.919616,149.4044,-35.139509 --config new_config.json --process new_process.lua`

The MBTiles was created perfectly, but on rendering it I discovered that nearby areas outside of the bounding box were rendered as well but only for lower zoom levels. So I was wondering -

1. What is the logic behind bbox on what gets rendered?
2. Is there some way in tilemaker such that only the area within bbox gets rendered or should I use osmium for that?

Thanks.

**Comment by systemed** [comment]
Basically all tiles that intersect the bbox will be rendered. At z14 this won't be much bigger than the bbox, but at (say) z5 the tiles are obviously pretty big, so any data contained in your .pbf for that tile will be written.

tilemaker doesn't have any clipping logic beyond this. You could indeed use osmium to achieve what you're looking for. That said, it wouldn't be daft to enhance tilemaker to skip objects outside the bbox when the `--bbox` argument is provided.

**Reply by kuwapa** [reply]
That makes sense. Thanks for the explanation. I'll use osmium to achieve the clipping behaviour.
